pkg/plugin/aggregation/run.go

```diff
@@ -40,6 +41,9 @@ const (
 	jitterFactor         = 1.2
 )
 
+// Variable for swapping during tests.
+var after = time.After
```

Comment:
In this particular case yes, I went with a more generic approach in case I have another case where I want to do something like channel shenanigans so that I can block/guarantee where I am in the flow or track how many times I hit the sleep with a counter.